cli/tools/script.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::colors;
+use crate::config_file::ConfigFile;
+use crate::flags::Flags;
+use crate::flags::ScriptFlags;
+use crate::proc_state::ProcState;
+use deno_core::anyhow::bail;
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use std::collections::HashMap;
+use std::process::Command;
+use std::sync::Arc;
+
+fn get_scripts_config(
+  maybe_config_file: Option<&ConfigFile>,
+) -> Result<HashMap<String, String>, AnyError> {
+  if let Some(config_file) = maybe_config_file {
+    let maybe_scripts_config = config_file.to_scripts_config()?;
+    if let Some(scripts_config) = maybe_scripts_config {
+      Ok(scripts_config)
+    } else {
+      bail!("No configured scripts")
+    }
+  } else {
+    bail!("No config file found")
+  }
+}
+
+fn print_available_scripts(scripts_config: HashMap<String, String>) {
+  eprintln!("{}", colors::green("Available scripts:"));
+
+  let mut script_names: Vec<String> =
+    scripts_config.clone().into_keys().collect();
+  script_names.sort();
+
+  for name in script_names {
+    eprintln!("- {}", colors::cyan(&name));
+    eprintln!("    {}", scripts_config[&name])
+  }
+}
+
+pub async fn execute_script(
+  flags: Flags,
+  script_flags: ScriptFlags,
+) -> Result<i32, AnyError> {
+  let flags = Arc::new(flags);
+  let ps = ProcState::build(flags.clone()).await?;
+  let scripts_config = get_scripts_config(ps.maybe_config_file.as_ref())?;
+  let config_file_url = &ps.maybe_config_file.as_ref().unwrap().specifier;
+  let config_file_path = if config_file_url.scheme() == "file" {
+    config_file_url.to_file_path().unwrap()
+  } else {
+    bail!("Only local configuration files are supported")
+  };
+
+  if script_flags.script.is_empty() {
+    print_available_scripts(scripts_config);
+    return Ok(1);
+  }
+
+  let cwd = config_file_path.parent().unwrap();
+  let script_name = script_flags.script;
+  let maybe_script = scripts_config.get(&script_name);
+
+  if let Some(script) = maybe_script {
+    let (shell_name, shell_flag) = if cfg!(windows) {
+      // TODO: allow to use PowerShell?
```

Comment:
Powershell is too different from sh.

Reply:
Good point, thanks

Reply:
Powershell Core does, but not Windows Powershell, which I believe most people will have installed on their system by default.

Reply:
There is no Core/windows only anymore under Powershell 7, which is the default on latest windows

Reply:
@lucsoft default on Windows 11? A lot of people are still using Windows 10 though. For me on Windows 10, the default seems to be Windows Powershell. I had to separately install Powershell 7. This doesn't sound like a great experience if many people are still on Windows Powershell by default. Regardless, I think we probably shouldn't do any of this because there's not a nice way to set environment variables in a cross platform way (see https://github.com/denoland/deno/pull/13725/files#r813337459).